xrpl/models/requests/ledger_entry.py

```diff
@@ -173,6 +172,33 @@ class Ticket(BaseModel):
     """
 
 
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class XChainClaimID(BaseModel):
+    """Required fields for requesting an XChainClaimID if not querying by object ID."""
+
+    locking_chain_door: str
+    locking_chain_issue: Currency
+    issuing_chain_door: str
+    issuing_chain_issue: Currency
+    xchain_claim_id: Union[int, str]
```

Comment:
Do these have to do the `= REQUIRED` thing we do for other interfaces? (Or no, because they're on a class?)

Reply:
The `= REQUIRED` syntax is only needed for transactions. This is because you can't have required params after optional params, so without the `= REQUIRED` type hack, you wouldn't be able to have transaction-specific required types, since there are some optional params in the `Transaction` base class.

Reply:
Good to know - thanks :)